util/logging.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package util
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/felixge/httpsnoop"
+	"github.com/gorilla/mux"
+	"go.elastic.co/ecszap"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Types of available loggers.
+const (
+	ECSLogger = "ecs"
+	DevLogger = "dev"
+
+	defaultLoggerType = ECSLogger
+)
+
+var logLevel = zap.LevelFlag("log-level", zap.InfoLevel, "log level (default \"info\")")
+var logType = flag.String("log-type", defaultLoggerType, "log type (ecs, dev)")
+
+var logger *zap.Logger
+var loggerMutex sync.Mutex
+
+// InitLogger initializes the logger, this is ignored afer a logger has
+// been created for this process.
+func InitLogger(loggerType string) {
```

Comment:
If we know that there are two types of loggers and no more, then we can replace them with `UseDevLogger` and `UseProductionLogger`.